src/foam/classloader/NodeModelFileDAO.js

```diff
@@ -21,36 +21,51 @@ foam.CLASS({
   extends: 'foam.dao.AbstractDAO',
 
   properties: [
-    {
-      name: 'classpath'
-    }
+    { class: 'String', name: 'classpath' },
+
+    { name: 'sep', factory: function() { return require('path').sep; } },
+    { name: 'fs',  factory: function() { return require('fs');       } },
+    { name: 'vm',  factory: function() { return require('vm');       } }
   ],
 
   methods: [
     function find(id) {
       var foamCLASS = foam.CLASS;
       var self = this;
-      var model;
+      var model = null;
 
       foam.CLASS = function(m) {
-        var cls = m.class ? foam.lookup(m.class) : foam.core.Model;
-        model = cls.create(m, self);
-        foam.CLASS = foamCLASS;
-      }
-
-      var sep = require('path').sep;
-      var path = this.classpath + sep + id.replace(/\./g, sep) + '.js';
-
-      try {
-        require(path);
-      } catch(e) {
-        console.warn('Unable to load at ' + path + '. Error: ' + e.stack);
-        return Promise.resolve(null);
-      } finally {
-        foam.CLASS = foamCLASS;
-      }
-
-      return Promise.resolve(model);
+        var cls = m.class ? foam.lookup(m.class) :
+              foam.core.Model;
+        var mdl = cls.create(m, self);
+        if ( mdl.id === id ) {
```

Comment:
If the file name matches, but it contains multiple `CLASS` calls (some of which may not be of the requested id).